src/templates/_select.twig

```diff
@@ -1,11 +1,126 @@
+            element.classList.add(`_extra_date_${right_now}`);
+        }
+    };
+
+    const attach_events = () => {
+        const commerce_wrapper = document.querySelector(
+            'div[class="{{ wrapper_class }}_{{ now }}"]'
+        );
+        add_date_class(commerce_wrapper);
+
+        const commerce_input = commerce_wrapper.querySelector(
+            'input[class="{{ filter_class }}_{{ now }}"]'
+        );
+        add_date_class(commerce_input);
+        const commerce_select = commerce_wrapper.querySelector(
+            'select[class="{{ name }}_{{ now }}"]'
+        );
+        add_date_class(commerce_select);
+
+        const commerce_select_array = [...commerce_select.options];
+
+        commerce_input.addEventListener('input', (event) => {
+            if (event.target.value) {
+                commerce_select_array.forEach((option) => {
+                    const regex = new RegExp(event.target.value, 'ig');
+                    if (regex.test(option.textContent)) {
+                        option.disabled = false;
+                        option.hidden = false;
+                    } else {
+                        option.disabled = true;
+                        option.hidden = true;
+                    }
+                });
+            } else {
+                commerce_select_array.forEach((option) => {
+                    option.disabled = false;
+                    option.hidden = false;
+                });
+            }
+        });
+
+        const commerce_selected_only = commerce_wrapper.querySelector(
+            'input[class="{{ selected_only_class }}_{{ now }}"]'
+        );
+        add_date_class(commerce_selected_only);
+
+        commerce_selected_only.addEventListener('change', (event) => {
+            if (event.target.checked) {
+                commerce_select_array.forEach((option) => {
+                    if (option.selected) {
+                        option.disabled = false;
+                        option.hidden = false;
+                    } else {
+                        option.disabled = true;
+                        option.hidden = true;
+                    }
+                });
+            } else {
+                commerce_select_array.forEach((option) => {
+                    option.disabled = false;
+                    option.hidden = false;
+                });
+            }
+        });
+
+        const commerce_clear_selected = commerce_wrapper.querySelector(
+            'button[class="{{ clear_selected_class }}_{{ now }}"]'
+        );
+        add_date_class(commerce_clear_selected);
+
+        commerce_clear_selected.addEventListener('click', (event) => {
+            event.preventDefault();
+            commerce_select_array.forEach((option) => {
+                if (option.selected) {
+                    option.selected = false;
+                }
+            });
+        });
+    };
+
+    attach_events();
```

Comment:
That would break the functionality. I suppose we can refactor it to use a different approach ... but this is already tested and works. I would need to run this through my QA team again and it would change the code quite a bit.